test/minimum_eigensolvers/evqe/test_evqe_individual.py

```diff
@@ -0,0 +1,297 @@
+# Quantum Evolving Ansatz Variational Solver (QUEASARS)
+# Copyright 2023 DLR - Deutsches Zentrum für Luft- und Raumfahrt e.V.
+
+from random import Random
+
+import pytest
+from pytest import raises
+from qiskit.circuit import QuantumCircuit
+
+from queasars.minimum_eigensolvers.evqe.evolutionary_algorithm.individual import (
+    RotationGate,
+    ControlGate,
+    ControlledRotationGate,
+    IdentityGate,
+    EVQECircuitLayer,
+    EVQECircuitLayerException,
+    EVQEIndividual,
+    EVQEIndividualException,
+)
+from queasars.utility.random import new_random_seed
+
+
+class TestEVQECircuitLayer:
+    def test_too_few_gates_is_invalid(self):
+        gates = (RotationGate(qubit_index=0), RotationGate(qubit_index=1))
+        with raises(EVQECircuitLayerException):
+            EVQECircuitLayer(n_qubits=3, gates=gates)
+
+    def test_too_many_gates_is_invalid(self):
+        gates = (RotationGate(qubit_index=0), RotationGate(qubit_index=1))
+        with raises(EVQECircuitLayerException):
+            EVQECircuitLayer(n_qubits=1, gates=gates)
+
+    def test_wrong_gate_ordering_is_invalid(self):
+        gates = (RotationGate(qubit_index=1), RotationGate(qubit_index=0))
+        with raises(EVQECircuitLayerException):
+            EVQECircuitLayer(n_qubits=2, gates=gates)
+
+    def test_gate_index_out_of_bounds_is_invalid(self):
+        gates = (RotationGate(-1), RotationGate(0), RotationGate(1))
+        with raises(EVQECircuitLayerException):
+            EVQECircuitLayer(n_qubits=3, gates=gates)
+
+        gates = (RotationGate(1), RotationGate(2), RotationGate(3))
+        with raises(EVQECircuitLayerException):
+            EVQECircuitLayer(n_qubits=3, gates=gates)
+
+    def test_loose_control_gate_is_invalid(self):
+        gates = (
+            ControlGate(qubit_index=0, controlled_qubit_index=2),
+            ControlGate(qubit_index=1, controlled_qubit_index=2),
+            ControlledRotationGate(qubit_index=2, control_qubit_index=1),
+        )
+        with raises(EVQECircuitLayerException):
+            EVQECircuitLayer(n_qubits=3, gates=gates)
+
+    def test_loose_controlled_rotation_gate_is_invalid(self):
+        gates = (
+            ControlledRotationGate(qubit_index=0, control_qubit_index=2),
+            ControlledRotationGate(qubit_index=1, control_qubit_index=2),
+            ControlGate(qubit_index=2, controlled_qubit_index=1),
+        )
+        with raises(EVQECircuitLayerException):
+            EVQECircuitLayer(n_qubits=3, gates=gates)
+
+    def test_random_layers_are_valid(self):
+        random_generator = Random(0)
+        for i in range(0, 100):
+            layer = EVQECircuitLayer.random_layer(
+                n_qubits=10, previous_layer=None, random_seed=new_random_seed(random_generator=random_generator)
+            )
+            assert layer.is_valid(), f"failed for random circuit layer number {i}"
```

Comment:
Please add small description of what failed: If later on one reads these messages it's easiert to get an immediate idea what was the problem.